src/cpuid/src/intel/leaves.rs

```diff
@@ -0,0 +1,1515 @@
+    0xBAu8 => "Data TLB1: 4 KByte pages, 4-way associative, 64 entries",
+    0xC0u8 => "Data TLB: 4 KByte and 4 MByte pages, 4-way associative, 8 entries",
+    0xC1u8 => "Shared 2nd-Level TLB: 4 KByte / 2 MByte pages, 8-way associative, 1024 entries",
+    0xC2u8 => "DTLB: 4 KByte/2 MByte pages, 4-way associative, 16 entries",
+    0xC3u8 => "Shared 2nd-Level TLB: 4 KByte / 2 MByte pages, 6-way associative, 1536 entries. Also 1GBbyte pages, 4-way, 16 entries.",
+    0xC4u8 => "DTLB: 2M/4M Byte pages, 4-way associative, 32 entries",
+    0xCAu8 => "Shared 2nd-Level TLB: 4 KByte pages, 4-way associative, 512 entries",
+    0xD0u8 => "3rd-level cache: 512 KByte, 4-way set associative, 64 byte line size",
+    0xD1u8 => "3rd-level cache: 1 MByte, 4-way set associative, 64 byte line size",
+    0xD2u8 => "3rd-level cache: 2 MByte, 4-way set associative, 64 byte line size",
+    0xD6u8 => "3rd-level cache: 1 MByte, 8-way set associative, 64 byte line size",
+    0xD7u8 => "3rd-level cache: 2 MByte, 8-way set associative, 64 byte line size",
+    0xD8u8 => "3rd-level cache: 4 MByte, 8-way set associative, 64 byte line size",
+    0xDCu8 => "3rd-level cache: 1.5 MByte, 12-way set associative, 64 byte line size",
+    0xDDu8 => "3rd-level cache: 3 MByte, 12-way set associative, 64 byte line size",
+    0xDEu8 => "3rd-level cache: 6 MByte, 12-way set associative, 64 byte line size",
+    0xE2u8 => "3rd-level cache: 2 MByte, 16-way set associative, 64 byte line size",
+    0xE3u8 => "3rd-level cache: 4 MByte, 16-way set associative, 64 byte line size",
+    0xE4u8 => "3rd-level cache: 8 MByte, 16-way set associative, 64 byte line size",
+    0xEAu8 => "3rd-level cache: 12MByte, 24-way set associative, 64 byte line size",
+    0xEBu8 => "3rd-level cache: 18MByte, 24-way set associative, 64 byte line size",
+    0xECu8 => "3rd-level cache: 24MByte, 24-way set associative, 64 byte line size",
+    0xF0u8 => "64-Byte prefetching",
+    0xF1u8 => "128-Byte prefetching",
+    0xFEu8 => "CPUID leaf 2 does not report TLB descriptor information; use CPUID leaf 18H to query TLB and other address translation parameters.",
+    0xFFu8 => "CPUID leaf 2 does not report cache descriptor information, use CPUID leaf 4 to query cache parameters"
+};
+impl fmt::Display for Leaf2 {
+    #[allow(clippy::unwrap_used, clippy::unwrap_in_result)]
+    #[inline]
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{:#?}", <[&'static str; 16]>::try_from(self).unwrap())
+    }
+}
+
+/// Error type for [`<[&'static str; 16] as TryFrom<&Leaf2>>::try_from`].
+#[derive(Debug, thiserror::Error, Eq, PartialEq)]
+#[error("Unknown cache and TLB infomation keyword: {0}")]
+pub struct UnknownKeyword(u8);
+
+// - The least-significant-byte of eax always returns 01h.
+// - The most significant bit indicates whether the register contains valid information (TODO Does
+//   this mean we only have 3 descriptors per register?)
+// TODO Implement the above conditions.
```

Comment:
There are various `TODOs` left in the code similar to this one. What's our plan to handle these?